go/arrow/csv/reader.go

```diff
@@ -982,7 +966,7 @@ func (c conversionColumn) inferType(v string) arrow.DataType {
 			c.typ = arrow.FixedWidthTypes.Boolean
 		case *arrow.BooleanType:
 			c.typ = arrow.FixedWidthTypes.Date32
-		case *arrow.Date32Type:
+		case *arrow.Date32Type, *arrow.Date64Type:
```

Comment:
since we never try to use `Date64Type` in this list of types for inferring, i don't think this gains us anything

Reply:
should I add it to the inferring somewhere? (maybe I missed something)

Reply:
The way the inferring works is essentially a loop that progressively tries the next type in that switch case. since both Date32 and Date64 expect the data to be in the same format it makes sense to only pick one for our inferring. 

So the question is which one do we want the default inferred type to be when it sees "YYYY-MM-DD" data? Date32 or Date64? whichever one we want the inferred default to be, that's the one we leave in this switch (and should be what `c.typ` is set to at line 968 above it

Reply:
date32 is fine then
b70d4706c844beae2e2bf2489627947baa1cd33c